services/accounts/accounts.go

```diff
@@ -72,23 +72,35 @@ func (api *API) GetAccounts(ctx context.Context) ([]*accounts.Account, error) {
 	return accounts, nil
 }
 
-func (api *API) DeleteAccount(ctx context.Context, address types.Address) error {
-	acc, err := api.db.GetAccountByAddress(address)
-	if err != nil {
-		return err
-	}
-	if acc.Type != accounts.AccountTypeWatch {
-		err = api.manager.DeleteAccount(address, "")
-		var e *account.ErrCannotLocateKeyFile
-		if err != nil && !errors.As(err, &e) {
+func (api *API) DeleteAccount(ctx context.Context, address types.Address, password string) error {
```

Comment:
Just for the context: why the account can not be deleted with the password?

Reply:
@rasom cause that's the way how delete account from the keystore function works, check it here: 
https://github.com/status-im/status-go/blob/develop/vendor/github.com/ethereum/go-ethereum/accounts/keystore/keystore.go#L244-L266

that function is called from here:
https://github.com/status-im/status-go/blob/develop/account/accounts.go#L609-L611

Without provided password we're not able to delete neither keystore file for that account nor keystore cache.

Reply:
ok, I se, thanks